### PR TITLE
Financial Connections: implemented new search footer

### PR DESCRIPTION
**Pull request by kgaidis-stripe**
## Summary

This PR implements a new search footer where showing manual entry is sometimes driven from the server:
- [Doc for this feature](https://docs.google.com/document/d/1nBrzQr2cxr3gWIjm13cdM9wOsm-NIsTWuyQ9BHbCSwc/edit#)
- [Designs](https://www.figma.com/file/nY1gZ4nuSmmDCUOysKSpzU/Connections---Auth-Flow-v2.1?node-id=9786-51341&t=KGhQsDIjvDUweB9K-0)

## Testing

We have to test two flows:
- Data flows (manual entry disabled)
- Payment flows (manual entry enabled)

And then there are also various search states:
- Nothing typed
- Many results (ex. 10 institutions)
- Some results (ex. 1 institution)
- No results
- Error

We also have to test whether the manual entry button is tappable (and navigates to next screen), and whether the manual entry disabled row (or no results / check spelling mistakes row) is _not_ tappable.

Below I captured & tested the above by using the iOS simulator.

### Manual Entry Enabled

| Designs (Many Results) | App (Many Results) |
| --- | ----------- | 
| Many results | Simulator Screen Shot - iPhone 14 Pro - 2023-03-23 at 15 19 52 |

| Designs (One Result) | App (One Result) |
| --- | ----------- | 
| One result (1) | Simulator Screen Shot - iPhone 14 Pro - 2023-03-23 at 15 20 00 |

| Designs (No Result) | App (No Result) |
| --- | ----------- | 
| No results (1) | Simulator Screen Shot - iPhone 14 Pro - 2023-03-23 at 15 20 03 |

| App (Nothing Typed) | App (Error) |
| --- | ----------- | 
| Simulator Screen Shot - iPhone 14 Pro - 2023-03-23 at 15 19 22 | Simulator Screen Shot - iPhone 14 Pro - 2023-03-23 at 15 27 20 |

### Navigating to Manual Entry Works

https://user-images.githubusercontent.com/105514761/227328895-d5d67a81-8382-4f20-a865-fd0b463cfed0.mp4

### Manual Entry Disabled

| Designs | No Results |
| --- | ----------- |
| No results | Simulator Screen Shot - iPhone 14 Pro - 2023-03-23 at 14 52 00  |

| Nothing Typed | Some Results | Error |
| ----------- | ----------- |  --- |
| Simulator Screen Shot - iPhone 14 Pro - 2023-03-23 at 14 51 37 | Simulator Screen Shot - iPhone 14 Pro - 2023-03-23 at 14 51 56 | Simulator Screen Shot - iPhone 14 Pro - 2023-03-23 at 15 08 23 |

### CODE REVIEW

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/InstitutionPicker/InstitutionSearchTableView.swift [hunk 1]
```diff
@@ -112,7 +132,7 @@ final class InstitutionSearchTableView: UIView {
             loadingView.trailingAnchor.constraint(equalTo: loadingContainerView.trailingAnchor),
         ])
 
-        showTableFooterView(true)
+        showTableFooterView(false)
```
Comment (vardges-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/InstitutionPicker/InstitutionSearchTableView.swift:135]: Why this change?

**Review by vardges-stripe — COMMENTED** [review]
Can we also have a test case with custom manual entry mode?

**Comment by kgaidis-stripe** [comment]
> Can we also have a test case with custom manual entry mode?

@vardges-stripe 

In what cases does custom manual entry mode affect the code changes from this PR? (ex. the decision to drop is handled by another layer)

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/InstitutionPicker/InstitutionSearchTableView.swift (hunk 1 above)
Comment (kgaidis-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Native/InstitutionPicker/InstitutionSearchTableView.swift:135]: @vardges-stripe 

good question as this also tripped my 'is this a bug' detector; 

This is in the initializer. Previously, the footer would nearly always show (except some cases where we stop showing it like errors I believe). Now, the footer only gets shown in specific cases.

**Review by vardges-stripe — APPROVED** [review]
Yep that's the video I wanted to see, essentially saying we still will call custom manual entry callback in this new design (I know we will just wanted baked in the test plan)
